src/tribler/core/components/restapi/rest/events_endpoint.py

```diff
@@ -144,7 +143,11 @@ async def _write_data(self, message: MessageDict):
             return
 
         for response in self.events_responses:
-            await response.write(message_bytes)
+            try:
+                await response.write(message_bytes)
+            except ConnectionResetError as e:
+                # The connection was closed by GUI
+                self._logger.warning(e, exc_info=True)
```

Comment:
I think we should also remove the `response` object with closed connection from `self.event_responses`. Note that we should not remove an item from a list during the iteration. To handle this, we can copy the list of event_responses before the loop:
```python
        for response in self.events_responses.copy():
            try:
                await response.write(message_bytes)
            except ConnectionResetError as e:
                # The connection was closed by GUI
                self._logger.warning(e, exc_info=True)
                self.events_responses.remove(response)
```

Reply:
Yes, you are right! I'll implement the suggestion a bit later.